src/coreclr/jit/emit.cpp

```diff
@@ -2834,6 +2838,18 @@ bool emitter::emitNoGChelper(CorInfoHelpFunc helpFunc)
 
         case CORINFO_HELP_INIT_PINVOKE_FRAME:
 
+        case CORINFO_HELP_FAIL_FAST:
+        case CORINFO_HELP_STACK_PROBE:
+
+        case CORINFO_HELP_CHECK_OBJ:
+
+        // never present on stack at the time of GC
+        case CORINFO_HELP_TAILCALL:
+
+        // called in prolog
+        case CORINFO_HELP_JIT_REVERSE_PINVOKE_ENTER:
+        case CORINFO_HELP_JIT_REVERSE_PINVOKE_ENTER_TRACK_TRANSITIONS:
```

Comment:
I don't think these are called in the prolog, so the comment doesn't seem accurate.

Reply:
The place where we use them says the helpers go into the prolog. 

https://github.com/dotnet/runtime/blob/4681f6318feef5d0d0bdba9f88876e266e65f2aa/src/coreclr/jit/flowgraph.cpp#L1689

Either way, these helpers do not need to be safe points (as in - we would not be unwinding through these calls and report what is alive). 

That avoids the problem of what GC info to emit for these - Anything that the containing method can report to GC is GC-dead before these calls and magically becomes alive after.

Reply:
I will just remove the confusing comment.